File: src/test/java/org/frawa/elmtest/run/ElmTestRunConfigurationTest.java
package org.frawa.elmtest.run;

import org.jdom.Element;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ElmTestRunConfigurationTest {

    @Test
    public void writeOptions() {
        Element root = new Element("ROOT");

        ElmTestRunConfiguration.Options options = new ElmTestRunConfiguration.Options();
        options.elmFolder = "folder";

        ElmTestRunConfiguration.writeOptions(options, root);

        assertEquals(1, root.getChildren().size());
        assertEquals(ElmTestRunConfiguration.class.getSimpleName(), root.getChildren().get(0).getName());
        assertEquals(1, root.getChildren().get(0).getAttributes().size());
        assertEquals("elm-folder", root.getChildren().get(0).getAttributes().get(0).getName());
        assertEquals("folder", root.getChildren().get(0).getAttributes().get(0).getValue());
    }

    @Test
    public void roundTrip() {
        Element root = new Element("ROOT");

        ElmTestRunConfiguration.Options options = new ElmTestRunConfiguration.Options();
        options.elmFolder = "folder";

        ElmTestRunConfiguration.writeOptions(options, root);
        ElmTestRunConfiguration.Options options2 = ElmTestRunConfiguration.readOptions(root);

        assertEquals(options.elmFolder, options2.elmFolder);
    }

}

File: src/test/kotlin/org/elm/ide/test/run/ElmTestRunConfigurationTest.kt
package org.elm.ide.test.run

import org.elm.ide.test.run.ElmTestRunConfiguration.Companion.readOptions
import org.elm.ide.test.run.ElmTestRunConfiguration.Companion.writeOptions
import org.elm.ide.test.run.ElmTestRunConfiguration.Options
import org.jdom.Element
import org.junit.Assert.assertEquals
import org.junit.Test

class ElmTestRunConfigurationTest {

    @Test
    fun writeOptions() {
        val root = Element("ROOT")

        val options = Options()
        options.elmFolder = "folder"

        writeOptions(options, root)

        assertEquals(1, root.children.size.toLong())
        assertEquals(ElmTestRunConfiguration::class.java.simpleName, root.children[0].name)
        assertEquals(1, root.children[0].attributes.size.toLong())
        assertEquals("elm-folder", root.children[0].attributes[0].name)
        assertEquals("folder", root.children[0].attributes[0].value)
    }

    @Test
    fun roundTrip() {
        val root = Element("ROOT")

        val options = Options()
        options.elmFolder = "folder"

        writeOptions(options, root)
        val options2 = readOptions(root)

        assertEquals(options.elmFolder, options2.elmFolder)
    }

}